extensions/amp-video/0.1/amp-video.js

```diff
@@ -192,6 +192,11 @@ class AmpVideo extends AMP.BaseElement {
     const {element} = this;
 
     this.configure_();
+    this.intersectionObserver_ = new IntersectionObserver(entries => {
+      const visible = entries[entries.length-1].isIntersecting;
+      this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible}); 
+    })
+    this.intersectionObserver_.observe(element);
```

Comment:
If we have an `observe`, we should also have an `unobserve`. Unfortunately right now we do not have the symmetric mount/dismount for the `buildCallback`, but we do it for `layout/unlayout`. But depending on what we need, we can provide a new pair of callbacks with some ease here.